aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1005,22 +1009,34 @@ std::tuple<Tensor, Tensor> _qr_helper_cpu(const Tensor& self, bool some) {
   return std::make_tuple(q_working_copy.narrow(-1, 0, n_columns_q), R);
 }
 
-std::tuple<Tensor,Tensor> qr(const Tensor& self, bool some) {
+std::tuple<Tensor,Tensor> linalg_qr(const Tensor& self, std::string mode) {
   TORCH_CHECK(self.dim() >= 2,
               "self should have at least 2 dimensions, but has ", self.dim(), " dimensions instead");
```

Comment:
"self" -> "qr input"

The name "self" is almost exclusively internal, and the error message should reflect the documentation, which says "input". The error message should also include the name of the function so people can understand where it's coming from.

Reply:
(I understand this is an existing problem, but while we're fixing qr we might as well improve it.)

Reply:
fixed by commit 8201ec9123 in PR #50046